errors/interceptor.go

```diff
@@ -0,0 +1,52 @@
+package errors
+
+import (
+	"context"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/grpclog"
+	"google.golang.org/grpc/status"
+)
+
+// UnaryServerInterceptor returns grpc.UnaryServerInterceptor
+// that should be used as a middleware to generate Error Messages
+// with Details and Field Information with Mapping given.
+func UnaryServerInterceptor(mapFuncs ...MapFunc) grpc.UnaryServerInterceptor {
+
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (res interface{}, err error) {
+
+		// Handle panic.
+		defer func() {
+			if perr := recover(); perr != nil {
+				err = status.Errorf(codes.Internal, "errors interceptor: %s", perr)
+				grpclog.Errorln(err)
+				res, err = nil, err
+			}
+		}()
+
+		// Initialize container with mapping.
+		container := InitContainer()
+		container.AddMapping(mapFuncs...)
+
+		// Save container in mapper.
+		ctx = NewContext(ctx, container)
```

Comment:
Should the comment be `Save container in context`?

Reply:
yes